VACMS-9387: update content Get Medical Records page - Aug 18 target merge

allisonlu opened the pull request:
trying this again... ([scrapped the previous PR](https://github.com/department-of-veterans-affairs/vets-website/pull/21510) due to something weird that happened when I merged `main` in)

**TARGET MERGE**: August 18th, for deploy on the 19th.

## Description
Updated FAQ copy on /health-care/get-medical-records/ with new content to reflect new Research & Support pages:

- [Get.your.medical.records_Cerner.version_for PW (1).docx](https://github.com/department-of-veterans-affairs/va.gov-cms/files/8970419/Get.your.medical.records_Cerner.version_for.PW.1.docx)
- [Get.your.medical.records_Non-Cerner.version_for.SME.team.docx](https://github.com/department-of-veterans-affairs/va.gov-cms/files/8970472/Get.your.medical.records_Non-Cerner.version_for.SME.team.docx)

## Original issue(s)
[/department-of-veterans-affairs/va.gov-cms/issues/9387](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/9387)

## Testing done
- [x] locally
- [ ] Get QA reviews from @RLHecht and @laurwill from Sitewide Content
- [ ] @jilladams to get test user for Cerner/authenticated content

## Screenshots

### Non-Cerner
![Screenshot of changes for Non-Cerner Get Medical Records page](https://user-images.githubusercontent.com/8542413/185462876-c7dc70b1-6671-4dfa-ad6d-ed6bf7cc26bb.png)

### Cerner
![Screenshot of changes for Non-Cerner Get Medical Records page](https://user-images.githubusercontent.com/8542413/185462990-7b933dc2-e786-4d43-bc20-105096784885.png)


## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

allisonlu committed:
VACMS-9387:
- updated removed "Please" from Sign In widget
- also changed "view" to "manage"

allisonlu committed:
VACMS-9387:
- updated content for both Unauth (non-Cerner) and Auth (Cerner) Get Medical Records pages
- includes: adding in accordions, updating link for DoD records, removed hard-coded instance of Mann-Grandstaff

allisonlu committed:
VACMS-9387:
- updated copy for logged-in Cerner user Sign In CTA

allisonlu commented:
@laurwill @RLHecht 

since we're not able to use test users on non-staging environments, here are links to the copy for the following use cases:

## logged-in Cerner user:
- https://github.com/department-of-veterans-affairs/vets-website/blob/5f0c85a9e28fb556baa062c6855d21bf43725f41/src/applications/static-pages/health-care-manage-benefits/components/CernerCallToAction/index.js#L167

## logged-in non-Cerner user:
- https://github.com/department-of-veterans-affairs/vets-website/blob/bf7b98d547d87e0322f9495f578ac6d733f0b227/src/applications/static-pages/cta-widget/components/messages/OpenMyHealtheVet.jsx#L11

laurwill commented:
Hi @allisonlu ! Thanks so much for providing all these screenshots and answering all my many questions. I have a few changes noted below — some of these are my original errors, so my apologies for that (Cerner differences continue to confuse the heck out of me). Let me know if you have any questions!

**Non-Cerner**
- In the signed-in grey box, remove "please" from "If you do, please sign in with the same account"
- Change accordion header “What if some of my medical records are missing online?” to “What if I can’t access all of my medical records through VA Blue Button?”
- In that same accordion section, change link text “Learn how to request medical records from your VA health facility” to “Learn how to get medical records from your VA health facility”
- In “Can I get notifications?” accordion, move period inside the quotation marks on “VA medical images and report available notification.”
- Make the 2 sign-in links in that same accordion hard returns rather than soft returns, to match other link formatting on page.
- In “How can I add information?” accordion, change “Go to the main navigation menu at the top of the page, and click on **Track Health.**” to “Go to the main navigation menu. Then select **Track Health.**”
- In “What if I have more questions?” accordion, include “TTY:” in the aria label hyperlink for the TTY number.

**Cerner**
- In “How can I add information?” accordion, change “Go to the main navigation menu at the top of the page, and click on Track Health.” to “Go to the main navigation menu. Then select Track Health.”
- In the “What if I can’t access all of my medical records?” accordion, change link text “Learn how to request medical records from your VA health facility” to “Learn how to get medical records from your VA health facility”
- In “What if I have more questions?” accordion, include “TTY:” in the aria label hyperlink for the TTY number.

allisonlu committed:
VACMS-9387:
- removed "please" from non-Cerner Sign In cta
- fixed copy updates for both Cerner and non-Cerner pages

allisonlu committed:
VACMS-9387:
- updated Telephone component to its corresponding web component so that TTY numbers are labelled so in its link

allisonlu committed:
VACMS-9387:
- removed redirect from MHV log in link

allisonlu committed:
VACMS-9387:
- remove redundant instance of "pop-up"

bkjohnson commented on src/applications/static-pages/health-care-manage-benefits/get-medical-records-page/components/AuthContent/index.js:
```diff
@@ -20,303 +17,242 @@ const AuthContent = ({
+            </li>
+            <li>
+              Call the My HealtheVet help desk at{' '}
+              <a href="tel:18773270022" aria-label="8 7 7. 3 2 7. 0 0 2 2.">
+                877-327-0022
+              </a>{' '}
```
This could be a `<va-telephone>`.

bkjohnson commented on src/applications/static-pages/health-care-manage-benefits/get-medical-records-page/components/AuthContent/index.js:
```diff
@@ -20,303 +17,242 @@ const AuthContent = ({
+          <h4>For My VA Health questions</h4>
+          <p>
+            Call My VA Health support anytime at{' '}
+            <a href="tel:18773270022" aria-label="8 0 0. 9 6 2. 1 0 2 4.">
+              800-962-1024
+            </a>
```
Same here. One advantage of using `<va-telephone>` with the `contact` prop is that there's no chance of mismatching numbers. This `tel` link has an 877 number but the actual text and the `aria-label` are showing an 800 number.

bkjohnson commented on src/applications/static-pages/health-care-manage-benefits/get-medical-records-page/components/UnauthContent/index.js:
```diff
@@ -59,194 +35,186 @@ const UnauthContent = () => (
+          </li>
+          <li>
+            Call the My HealtheVet help desk at{' '}
+            <a href="tel:18773270022" aria-label="8 7 7. 3 2 7. 0 0 2 2.">
+              877-327-0022
+            </a>{' '}
```
Another opportunity for a `<va-telephone>`.

bkjohnson reviewed (commented):
The ESLint warnings should be straightforward to resolve:

```diff
diff --git a/src/applications/static-pages/health-care-manage-benefits/components/CernerCallToAction/index.js b/src/applications/static-pages/health-care-manage-benefits/components/CernerCallToAction/index.js
index 7762ca354d..f738d6b049 100644
--- a/src/applications/static-pages/health-care-manage-benefits/components/CernerCallToAction/index.js
+++ b/src/applications/static-pages/health-care-manage-benefits/components/CernerCallToAction/index.js
@@ -5,7 +5,6 @@ import { isEmpty, map, replace } from 'lodash';
 import * as Sentry from '@sentry/browser';
 // Relative imports.
 import AlertBox from '@department-of-veterans-affairs/component-library/AlertBox';
-import LoadingIndicator from '@department-of-veterans-affairs/component-library/LoadingIndicator';
 import environment from 'platform/utilities/environment';
 import recordEvent from 'platform/monitoring/record-event';
 import { apiRequest } from 'platform/utilities/api';
@@ -95,7 +94,7 @@ export class CernerCallToAction extends Component {
     if (fetching) {
       return (
         <div data-testid="cerner-cta-widget">
-          <LoadingIndicator message="Loading your information..." />
+          <va-loading-indicator message="Loading your information..." />
         </div>
       );
     }
diff --git a/src/applications/static-pages/health-care-manage-benefits/components/CernerCallToAction/index.unit.spec.js b/src/applications/static-pages/health-care-manage-benefits/components/CernerCallToAction/index.unit.spec.js
index 908a209563..5ba4a21b75 100644
--- a/src/applications/static-pages/health-care-manage-benefits/components/CernerCallToAction/index.unit.spec.js
+++ b/src/applications/static-pages/health-care-manage-benefits/components/CernerCallToAction/index.unit.spec.js
@@ -9,8 +9,8 @@ describe('<CernerCallToAction>', () => {
   it('renders what we expect on its initial render', () => {
     const wrapper = shallow(<CernerCallToAction />);
 
-    const text = wrapper.text();
-    expect(text).to.include('<LoadingIndicator />');
+    const text = wrapper.html();
+    expect(text).to.include('<va-loading-indicator />');
 
     wrapper.unmount();
   });
```

laurwill commented:
Hi @allisonlu , noting some last fixes here from my recheck of the files:
- In signed-in Cerner sign-in box, remove extra "pop-up" from third sentence and make sure second sentence reads "If you do, sign in with the same account you used to sign in here on VA.gov."
- In Cerner "What you can do in My HealtheVet's VA Blue Button" section, change "medications" to "medicines" in second bullet
- In Non-Cerner "Can I get notifications" section, remove redirect from this URL https://www.myhealth.va.gov/mhv-portal-web/web/myhealthevet/user-login 
- In both versions "What if I have more questions" section, should the full TTY phone number also be showing here? 
<img width="493" alt="image" src="https://user-images.githubusercontent.com/104158104/185671360-f7598cf7-8d99-416d-874b-b3422be1a249.png">

Thank you!

obliviga commented:
Aside from Brooks's comments, this LGTM!

allisonlu committed:
VACMS-9387: migrated all telephone links to use va-telephone web component

allisonlu commented:
Thanks @bkjohnson, I'll address migrating the React components to web components in another PR because we're trying to get this content out with today's deploy

allisonlu commented:
@laurwill 

- [x] In signed-in Cerner sign-in box, remove extra "pop-up" from third sentence and make sure second sentence reads "If you do, sign in with the same account you used to sign in here on VA.gov."
- [x] In Cerner "What you can do in My HealtheVet's VA Blue Button" section, change "medications" to "medicines" in second bullet
- [x] In Non-Cerner "Can I get notifications" section, remove redirect from this URL https://www.myhealth.va.gov/mhv-portal-web/web/myhealthevet/user-login
- [x] In both versions "What if I have more questions" section, should the full TTY phone number also be showing here?

this is how the number appears on the page. the number is stored as a constant 
![Screen Shot 2022-08-19 at 1 43 07 PM](https://user-images.githubusercontent.com/8542413/185676986-af3777e9-e569-4217-8408-981bb34336ee.png)

ryguyk commented on src/applications/static-pages/cta-widget/components/messages/OpenMyHealtheVet.jsx:
```diff
@@ -12,9 +12,9 @@ const OpenMyHealtheVet = ({
-        {toolName} tool. If you do, please sign in with the same account you
-        used to sign in here on VA.gov. You also may need to disable your
-        browser’s pop-up blocker so that My HealtheVet will be able to open.
+        {toolName} tool. If you do, sign in with the same account you here on
```
This feels like a typo.

ryguyk replied:
This was fixed.

allisonlu committed:
VACMS-9387: minor copy changes

allisonlu committed:
updated unit tests

allisonlu committed:
update unit test for CernerCTA
